### PR TITLE
Slow log fields

### PR DESCRIPTION
**Pull request by breeswish**
Signed-off-by: Breezewish <breezewish@pingcap.com>

## What have you changed? (mandatory)

This PR introduces fields for Coprocessor slow log.

## What are the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

- [x] Unit test
- [x] Manual integration test for the slow log

  Current log output:

  ```
   2019/01/15 23:33:45.507 INFO tracker.rs:175: slow-query, internal_key_skipped_count: 26357, internal_delete_skipped_count: 8786, block_cache_hit_count: 4, block_read_count: 0, block_read_byte: 0, scan_first_range: [74800000000000001F5F72D73ED3CDD7EB8C8D, 74800000000000001F5F72FFFFFFFFFFFFFFFF00), scan_ranges: 1, scan_iter_processed: 17572, scan_iter_ops: 17574, scan_is_desc: false, tag: select, table_id: 31, txn_start_ts: 405685251798728705, wait_time: 1.683s, total_process_time: 1.103s, peer_id: ipv4:127.0.0.1:58482, region_id: 2
   ```

## Does this PR affect documentation (docs) update? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

### CODE REVIEW

File: components/log_wrappers/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+extern crate slog;
+
+// We don't introduce namespace std for std functions.
+
+pub mod time {
+    pub struct Duration<'a>(pub &'a ::std::time::Duration);
+
+    impl<'a> ::slog::Value for Duration<'a> {
+        #[inline]
+        fn serialize(
```
[the remark this answers is not in the document]
Comment (breeswish) [reply components/log_wrappers/src/lib.rs]: Yes. Work in progress ⭐️

[the remark this answers is not in the document]
Comment (breeswish) [reply components/log_wrappers/src/lib.rs]: Unit tests added! 😄

File: Cargo.toml [hunk 2]
```diff
@@ -101,29 +102,26 @@ num = "0.2"
...
 smallvec = { version = "0.6", features = ["union"] }
 flate2 = { version = "1.0", features = ["zlib"], default-features = false }
 more-asserts = "0.1"
 hyper = "0.12"
 tokio-threadpool = "0.1"
 vlog = "0.1.4"
+cop_datatype = { path = "components/cop_datatype" }
+panic_hook = { path = "components/panic_hook" }
+tipb = { path = "components/tipb" }
+kvproto = { path = "components/kvproto" }
```
[the remark this answers is not in the document]
Comment (AndreMouche) [reply Cargo.toml]: Seems he wants to  implement  `log::Value` for KeyRange in kvproto

File: components/log_wrappers/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+extern crate kvproto as lib_kvproto;
+#[macro_use]
+extern crate slog;
+extern crate slog_term;
+
+pub mod test_util;
+
+// We don't introduce namespace std for std functions.
+
+pub mod time {
```
Comment (Hijiao) [inline components/log_wrappers/src/lib.rs]: As a result, you can also remove this mod.

File: Cargo.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hijiao) [reply Cargo.toml]: Is it easier to implement `fmt::Display` trait for KeyRange? @breeswish

[the remark this answers is not in the document]
Comment (breeswish) [reply Cargo.toml]: @Hijiao  We cannot implement fmt::Display for KeyRange

File: Cargo.toml [hunk 4]
```diff
@@ -101,29 +102,26 @@ num = "0.2"
...
-cop_datatype = { path = "components/cop_datatype" }
 smallvec = { version = "0.6", features = ["union"] }
 flate2 = { version = "1.0", features = ["zlib"], default-features = false }
 more-asserts = "0.1"
 hyper = "0.12"
 tokio-threadpool = "0.1"
 vlog = "0.1.4"
+cop_datatype = { path = "components/cop_datatype" }
+panic_hook = { path = "components/panic_hook" }
+tipb = { path = "components/tipb" }
```
[the remark this answers is not in the document]
Comment (breeswish) [reply Cargo.toml]: https://github.com/tikv/tikv/issues/3887

File: components/log_wrappers/src/lib.rs (hunk 3 above)
Comment (breeswish) [reply components/log_wrappers/src/lib.rs]: Reserved for supporting `Option<Duration>`

Comment (breeswish) [reply components/log_wrappers/src/lib.rs]: Updated, make it generic so that it can be used with any `T: Display` or `T: Debug`.

**Comment by siddontang** [comment]
how about the new output?

**Comment by breeswish** [comment]
@Hijiao @AndreMouche Please give an approval, thanks~

**Comment by Hijiao** [comment]
Oh, please fix the conflicting file.

**Pull request merged**